File: 59-spiral-matrix-ii/spiral-matrix-ii.cs
public class Solution {
    public int[][] GenerateMatrix(int n) {
        int[][] matrix = new int[n][];
        for (int i = 0; i < n; i++) {
            matrix[i] = new int[n];
        }
        int num = 1, top = 0, left = 0, bottom = n - 1, right = n - 1;
        while (num <= n * n) {
            for (int i = left; i <= right; i++)
                matrix[top][i] = num++;
            top++;
            for (int i = top; i <= bottom; i++)
                matrix[i][right] = num++;
            right--;
            for (int i = right; i >= left; i--)
                matrix[bottom][i] = num++;
            bottom--;
            for (int i = bottom; i >= top; i--)
                matrix[i][left] = num++;
            left++;
        }
        return matrix;
    }
}

File: 59-spiral-matrix-ii/spiral-matrix-ii.rs
impl Solution {
    pub fn generate_matrix(n: i32) -> Vec<Vec<i32>> {
        let n = n as usize;
        let mut matrix = vec![vec![0; n]; n];
        let mut num = 1_i32;
        let mut top: isize = 0;
        let mut left: isize = 0;
        let mut bottom: isize = (n as isize) - 1;
        let mut right: isize = (n as isize) - 1;

        while num <= (n * n) as i32 {
            for i in left..=right {
                matrix[top as usize][i as usize] = num;
                num += 1;
            }
            top += 1;
            for i in top..=bottom {
                matrix[i as usize][right as usize] = num;
                num += 1;
            }
            right -= 1;
            for i in (left..=right).rev() {
                matrix[bottom as usize][i as usize] = num;
                num += 1;
            }
            bottom -= 1;
            for i in (top..=bottom).rev() {
                matrix[i as usize][left as usize] = num;
                num += 1;
            }
            left += 1;
        }

        matrix
    }
}
